.github/workflows/check-docs-build.yml

```diff
@@ -1,9 +1,11 @@
 name: check docs build
 
 on:
+  push:
+    branches: [ "main" ]
```

Comment:
What do we gain from this? Won't we just be repeating the tests that were run for the PR?

Reply:
This was more to bring the C.I. checks in line with what happens on other repositories. Unless I'm mistaken the docs build is tested for other pytket repos when commits are pushed to a branch in a PR and also when the PR is merged. At present only the former happens for pytket-docs.

Reply:
Looks like for pytket it is only run for `pull_request` and `workflow_dispatch`. https://github.com/CQCL/tket/blob/75d584f615d0de8129eb8f147222505f0ebdb2de/.github/workflows/build_and_test.yml#L315

It may be different on other repos, in which case perhaps we should change that. I don't see any point in running the check twice on exactly the same code...

Reply:
I see. Looking at the workflow for the extensions it seems it works in the way I thought?
https://github.com/CQCL/pytket-quantinuum/blob/main/.github/workflows/build_and_test.yml

I don't have a strong opinion either way... was only doing this for consistency. Fine with closing this if its unecessary.

Reply:
Even on that one the docs build is only run for `pull_request` and `schedule` triggers: https://github.com/CQCL/pytket-quantinuum/blob/42cd71c7d581c6c94bd3cde759673ce73c18a3a1/.github/workflows/build_and_test.yml#L148

For other unit tests I think we run using different Python versions on push to main, which is a compromise between thoroughness and saving time. But for the docs build I assume we only need to test with one Python version.